src/System.Private.ServiceModel/tools/test/Bridge/Bridge/ConfigController.cs

```diff
@@ -44,16 +55,24 @@ public HttpResponseMessage POST(HttpRequestMessage request)
                 string nameValuePairs = request.Content.ReadAsStringAsync().Result;
                 Dictionary<string, string> configInfo = JsonSerializer.DeserializeDictionary(nameValuePairs);
 
+                Trace.WriteLine("POST config content: " + Environment.NewLine + nameValuePairs);
+
                 // Create a new configuration combining the existing one with any provided properties.
                 BridgeConfiguration newConfiguration = new BridgeConfiguration(BridgeConfiguration, configInfo);
+                Trace.WriteLine("POST config properties: " + Environment.NewLine + newConfiguration);
 
-                Trace.WriteLine("POST config request: " + Environment.NewLine + newConfiguration);
+                // Take the new configuration and notify listeners of the change.
+                BridgeConfiguration oldConfiguration = BridgeConfiguration;
+                BridgeConfiguration = newConfiguration;
 
-                string friendlyName = newConfiguration.UpdateApp();
-                CurrentAppDomain = friendlyName;
+                if (BridgeConfigurationChanged != null)
+                {
+                    Tuple<BridgeConfiguration, BridgeConfiguration> eventArgs = new Tuple<BridgeConfiguration, BridgeConfiguration>(oldConfiguration, newConfiguration);
+                    BridgeConfigurationChanged(this, eventArgs);
+                }
 
-                string configResponse = PrepareConfigResponse(TypeCache.Cache[friendlyName]);
-                Trace.WriteLine("POST config response: " + Environment.NewLine + configResponse);
+                string configResponse = PrepareConfigResponse(TypeCache.Cache[CurrentAppDomainName]);
+                Trace.WriteLine("POST config response content: " + Environment.NewLine + configResponse);
```

Comment:
Is there any benefit to using the Category parameter in tracing? I could see some benefit in having the class name provided as the category.